File: src/meihuayishu/liuqin.go
package meihuayishu

type LiuQin struct {
	Id int
}

const (
	FUMU    = iota // 父母
	ZISUN          // 子孫
	GUANGUI        // 官鬼
	QICAI          // 妻财
	XIONGDI        // 兄弟
)

func (lq LiuQin) GetName() string {
	switch lq.Id {
	case 0:
		return "父母"
	case 1:
		return "子孙"
	case 2:
		return "官鬼"
	case 3:
		return "妻财"
	case 4:
		return "兄弟"
	}
	return ""
}

File: kotlin/src/main/kotlin/com/github/lusing/liuyao/LiuQin.kt
package com.github.lusing.liuyao

class LiuQin {
    var lq: Int = XIONGDI

    fun getName() :String {
        when  (this.lq) {
            FUMU->
            return "父母"
            ZISUN->
            return "子孙"
            GUANGUI->
            return "官鬼"
            QICAI->
            return "妻财"
            XIONGDI->
            return "兄弟"
        }
        return ""
    }

    companion object {
        const val FUMU = 0    // 父母
        const val ZISUN = 1          // 子孫
        const val GUANGUI = 2        // 官鬼
        const val QICAI = 3          // 妻财
        const val XIONGDI = 4        // 兄弟
    }
}



